### ISSUE TITLE
compute_iou()?

### ISSUE THREAD

**Issue by sunsunnyshine** [SysCV/sam-hq#80]
The input parameters consist of a 256x256 network output and the ground truth (GT). Has the post-processing of the output already been performed in DecoderHQ? If not, doesn't this mean that they are not aligned?

**Comment by ymq2017** [comment]
Hi, there are two ways to align them.
1. in training, [compute_iou function](https://github.com/SysCV/sam-hq/blob/main/train/train.py#L505) will align the shape of prediction to original input.
2. in inference, you can also use [postprocess function](https://github.com/SysCV/sam-hq/blob/main/segment_anything/modeling/sam.py#L136). And it is already used in [predictor class](https://github.com/SysCV/sam-hq/blob/main/segment_anything/predictor.py#L245)

**Comment by sunsunnyshine** [comment]
Thank you for your response!

However, I believe post-processing should involve removing the padding region first before resizing. I didn't see any handling of removing padding for 'preds' in the 'compute_iou()' function. I have some concerns.

**Comment by ymq2017** [comment]
Yes if the image is loaded with padding, removing the padding region is needed. And I would recommend using postprocess function for inference.
In training for compute_iou, we do not use padding. you can refer to [this line](https://github.com/SysCV/sam-hq/blob/main/train/train.py#L339) and the [resize function](https://github.com/SysCV/sam-hq/blob/main/train/utils/dataloader.py#L103).

**Comment by sunsunnyshine** [comment]
I got it, thank you for your patient response!
